lib/omniauth/strategies/open_id.rb

```diff
@@ -59,13 +60,21 @@ def start
         end
       end
 
-      def get_identifier
+      def openid_form
         f = OmniAuth::Form.new(:title => 'OpenID Authentication')
         f.label_field('OpenID Identifier', options.identifier_param)
         f.input_field('url', options.identifier_param)
         f.to_response
       end
 
+      def get_identifier
+        if options[:login_entrypoint].respond_to?(:call)
+          instance_eval &options[:login_entrypoint]
```

Comment:
Because if so we won't be able to use something like `fail!(:THE_FAILURE)` in the omniauth initializer.